### PR TITLE
Use Javascript onChange for a much more elegant map marker solution

### PR DESCRIPTION
**Pull request by AmberWilkie**
Changes proposed in this pull request:
- Implement Javascript solution to reloading the map
- Remove `dropdown` route and method

What I have learned working on this feature:
- Javascript variables... but mostly, this was a logic exercise. How to get information from one place to another.

PS: I shifted the geoLocation function over a tab, but I didn't actually change any code. 

Ready for review!

### CODE REVIEW

**Comment by AmberWilkie** [comment]
Ok coveralls, that's just not fair. I removed code and now the percentage is lower? The build reads that we're testing 100% of every file we're testing at all, so this is just a ratio thing, right? I used to have more lines of code to test, and now I've condensed things and my numbers go down?

File: app/views/application/_category_drop.html.haml [hunk 1]
```diff
@@ -1,2 +1,2 @@
-= form_tag(controller: :restaurants, action: :dropdown, class: 'form-group') do
-  = select_tag 'category', options_for_select(dropdown_options), include_blank: 'Pick a Category', onChange: "javascript: $('form').submit()", class: 'form-control'
+= form_tag(class: 'form-group') do
```
Comment (tochman) [inline app/views/application/_category_drop.html.haml]: You don't need this form anymore.

Comment (tochman) [reply app/views/application/_category_drop.html.haml]: Just keep the select_tag, or is that not possible?

Comment (tochman) [reply app/views/application/_category_drop.html.haml]: Not sure about this

Comment (AmberWilkie) [reply app/views/application/_category_drop.html.haml]: seems to work fine

**Comment by tochman** [comment]
Any objections to this one @diraulo We are dropping in test coverage without Coveralls telling us why...

**Comment by diraulo** [comment]
@tochman 


My take on this is the removed lines had an tiny impact on the coverage calculations. One thing i would suggest is to set the coverage decrease threshold for failure on coveralls to 1% or 0.5%

**Comment by tochman** [comment]
I can't find where those settings are being set @diraulo They changed their interface and I just can't locate those settings. I remember there was a form for that before on Coveralls.

**Comment by tochman** [comment]
Found it...

**Pull request merged**